lettuce/native_generator/_generator.py

```diff
@@ -0,0 +1,209 @@
+        val['python_wrapper_before_buffer'] = t1.join(self.buf['python_wrapper_before'])
+        val['python_wrapper_after_buffer'] = t1.join(self.buf['python_wrapper_after'])
+
+        val['kernel_parameter'] = c1.join(self.par['kernel_parameter_name'])
+        val['kernel_parameter_values'] = c1.join(self.par['kernel_parameter_value'])
+
+        val['cuda_wrapper_parameter'] = c1.join(self.par['cuda_wrapper_parameter_name'])
+        val['cuda_wrapper_parameter_values'] = c1.join(self.par['cuda_wrapper_parameter_value'])
+
+        val['cpp_wrapper_parameter'] = c1.join(self.par['cpp_wrapper_parameter_name'])
+        val['cpp_wrapper_parameter_value'] = c1.join(self.par['cpp_wrapper_parameter_value'])
+
+        # release resources
+        self.reset()
+        return val
+
+    @staticmethod
+    def format(val) -> str:
+        import os.path
+        import shutil
+        import tempfile
+
+        def fmt(text):
+            return text.format(**val)
+
+        input_root_dir = os.path.abspath(os.path.join(__file__, '..', 'template'))
+        output_root_dir = tempfile.mkdtemp()
+
+        for (sub_root, sub_dirs, sub_files) in os.walk(input_root_dir):
+
+            input_sub_root = os.path.relpath(sub_root, input_root_dir)
+            output_sub_root = os.path.relpath(fmt(sub_root), input_root_dir)
+
+            if '__pycache__' in output_sub_root:
+                continue
+
+            if os.path.isdir(os.path.join(output_root_dir, output_sub_root)):
+                shutil.rmtree(os.path.join(output_root_dir, output_sub_root), ignore_errors=True)
+            os.makedirs(os.path.join(output_root_dir, output_sub_root), exist_ok=False)
+
+            for sub_file in sub_files:
+                input_sub_file = os.path.join(input_root_dir, input_sub_root, sub_file)
+                output_sub_file = os.path.join(output_root_dir, output_sub_root, fmt(sub_file))
+
+                with open(input_sub_file, 'r') as input_sub_file:
+                    with open(output_sub_file, 'w') as output_sub_file:
+                        tmp = input_sub_file.read()
+                        tmp = fmt(tmp)
+                        output_sub_file.write(tmp)
+
+        return output_root_dir
+
+    def resolve(self):
+        try:
+            import importlib
+            native = importlib.import_module(f"lettuce_native_{self.name}")
+            # noinspection PyUnresolvedReferences
+            return native.stream_and_collide
+        except ModuleNotFoundError:
+            # TODO log error? 'combination not yet generated!'
+            return None
+        except AttributeError:
+            # TODO log error? 'combination is not installed properly!'
+            return None
+
+    @staticmethod
+    def install(directory: str):
+        import os.path
+        import subprocess
+
+        setup_file = os.path.join(directory, 'setup.py')
+        subprocess.call(['python', setup_file, 'install'], shell=True, cwd=directory)
```

Comment:
This should install to the same prefix as this file is in, e.g. `--prefix`, `os.path.join(os.path.dirname(__file__), "../../" )`